qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateEnumAnnotationSupport.java

```diff
@@ -0,0 +1,116 @@
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.jdt.internal.template.datamodel;
+
+import static com.redhat.qute.jdt.internal.QuteJavaConstants.TEMPLATE_ENUM_ANNOTATION;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IAnnotatable;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.qute.commons.datamodel.resolvers.ValueResolverInfo;
+import com.redhat.qute.jdt.QuteSupportForTemplate;
+import com.redhat.qute.jdt.internal.resolver.ITypeResolver;
+import com.redhat.qute.jdt.template.datamodel.AbstractAnnotationTypeReferenceDataModelProvider;
+import com.redhat.qute.jdt.template.datamodel.SearchContext;
+import com.redhat.qute.jdt.utils.AnnotationUtils;
+
+/**
+ * @TemplateEnum annotation support.
+ *
+ *
+ *               <code>
+ * &#64;TemplateEnum
+ * public enum Status {
+ * 		ON,
+ * 		OFF
+ * }
+ * </code>
+ *
+ * @see https://quarkus.io/guides/qute-reference#convenient-annotation-for-enums
+ *
+ */
+public class TemplateEnumAnnotationSupport extends AbstractAnnotationTypeReferenceDataModelProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateEnumAnnotationSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+			TEMPLATE_ENUM_ANNOTATION
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+			SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (javaElement.getElementType() != IJavaElement.TYPE) {
+			return;
+		}
+		IType type = (IType) javaElement;
+		IAnnotation templateEnum = AnnotationUtils.getAnnotation((IAnnotatable) javaElement, TEMPLATE_ENUM_ANNOTATION);
+		if (templateEnum == null) {
+			return;
+		}
+
+		if (javaElement instanceof IType) {
+			IType type = (IType) javaElement;
+			collectResolversForTemplateEnum(type, templateEnum, context.getDataModelProject().getValueResolvers(),
+					monitor);
+		}
+	}
+
+	private static void collectResolversForTemplateEnum(IType type, IAnnotation templateExtension,
+			List<ValueResolverInfo> resolvers, IProgressMonitor monitor) {
+		try {
+			ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(type);
+			if (type.isEnum()) {
+				IField[] fields = type.getFields();
+				for (IField field : fields) {
+					IAnnotation fieldFromTemplateEnum = AnnotationUtils.getAnnotation(field, TEMPLATE_ENUM_ANNOTATION);

```

Comment:
Is field can be annotated with@TemplateEnum ?

Reply:
As per the reference guide example: 
```java
public enum Status {
    ON,
    OFF
}
```
What are `ON` and `OFF` considered?

Reply:
ON and OFF are fields. I think you cannot write:

@TemplateEnum
ON

Reply:
In other words replace with this code:

```
IField[] fields = type.getFields();
for (IField field : fields) {
    collectResolversForTemplateEnum(field, resolvers, typeResolver) {
}
```